src/imars3d/backend/io/metadata.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+"""
+Metadata class for IMars3D
+"""
+import numpy as np
+import param
+import tifffile
+from pathlib import Path
+
+
+# setup module level logger
+logger = param.get_logger(__name__)
+logger.name = __name__
+
+
+class MetaData(param.Parameterized):
+    """
+    Metadata extracted from given file
+    """
+
+    filename = param.Filename(doc="full path to the file.")
+    datatype = param.Selector(
+        default="ct",
+        objects=[
+            "radiograph",
+            "ct",
+            "openbeam",
+            "ob",
+            "darkcurrent",
+            "dc",
+        ],
+        doc="data type [ct|ob|dc]",
+    )
+    metadata_index = param.List(
+        default=[
+            65026,  # "ManufacturerStr:Andor"
+            65027,  # "ExposureTime:70.000000"
+            65068,  # "MotSlitHR.RBV:10.000000"
+            65070,  # "MotSlitHL.RBV:20.000000"
+            65066,  # "MotSlitVT.RBV:10.000000"
+            65068,  # "MotSlitHR.RBV:10.000000"
+        ],
+        doc="list of metadata keys to be extracted from the file.",
+    )
+    metadata = param.Dict(
+        default={},
+        doc="dictionary of metadata extracted from the file.",
+    )
+    relative_tolerance = param.Number(
+        default=0.01,
+        doc="relative tolerance for comparing metadata values.",
+    )
+
+    @property
+    def suffix(self) -> str:
+        return Path(self.filename).suffix
+
+    @param.depends("filename", "datatype", watch=True, on_init=True)
+    def _update_metadata_index(self):
+        """
+        Update metadata index
+        """
+        if self.suffix.lower() in (".tiff", "tif"):
+            # NOTE: trim slits position from metadata_keys if the file is
+            #       dark current file
+            if self.datatype in ("darkcurrent", "dc"):
+                self.metadata_index = [65026, 65027]
+        elif self.suffix.lower() in (".fits", ".fit"):
+            raise NotImplementedError
```

Comment:
```suggestion
            raise ValueError(f'Suffix="{self.suffix}" is not currently supported')
```
I'm ok with `NotImplementedError` too, but you should tell the caller what isn't supported. It will make a much faster investigation when people try to use it on a different kind of file. This should be done with the other places in `MetaData` that throw an empty exception.